enterprise/models.py

```diff
@@ -2849,6 +2849,25 @@ class SystemWideEnterpriseUserRoleAssignment(UserRoleAssignment):
     class Meta:
         unique_together = (('enterprise_customer', 'role', 'user'),)
 
+    def _validate_whether_enterprise_customer_needed(self):
+        """
+        Verify that if enterprise_customer is None, then has_access_to_all_contexts
+        must be true.
+
+        Raises ValidationError
+        """
+
+        if self.enterprise_customer is None and self.has_access_to_all_contexts() is not True:
+            message = (
+                "Enterprise customer must be set on SystemWideEnterpriseUserRoleAssignment "
+                f"instance {self.id} because has_access_to_all_contexts is not True."
+            )
+            raise ValidationError(message)
+
+    def save(self, *args, **kwargs):
+        self._validate_whether_enterprise_customer_needed()
```

Comment:
I'm worried that this exception message is going to end up getting swallowed and logged in a strange way... Should I be concerned? Would it be better to just put the message to the logger in the `_validate_whether_enterprise_customer_needed` method? 

also this save method now also raises `ValidationError`.

Reply:
I can add a log line for sure 👍 

For what it's worth, models' save() method already raises ValidationError if a field is not correct (like if you try to set a bool field to a string)